src/components/Popup/Popup.tsx

```diff
@@ -126,16 +127,45 @@ export default class Popup extends AutoControlledComponent<Extendable<PopupProps
 
   private static isBrowserContext = isBrowser()
 
+  private clickSubscription = EventStackSubscription.empty()
+  private triggerDomElement = null
+
   protected actionHandlers: AccessibilityActionHandlers = {
-    toggle: e => this.trySetOpen(!this.state.open, e, true),
-    closeAndFocusTrigger: e => {
-      this.trySetOpen(false, e, true)
-      _.invoke(this.state.target, 'focus')
+    toggle: e => {
+      this.trySetOpen(!this.state.open, e, true)
     },
+    closeAndFocusTrigger: e => this.closeAndFocusTrigger(e),
+  }
+
+  private closeAndFocusTrigger = e => {
+    if (this.state.open) {
+      this.trySetOpen(false, e, true)
+      _.invoke(this.triggerDomElement, 'focus')
+    }
+  }
+
+  private closeAndFocusTriggerOnClickIfOpen() {
+    if (this.state.open) {
+      setTimeout(() => this.clickSubscription.replaceWith('click', this.closeAndFocusTrigger))
```

Comment:
lets start with the simplest thing - specifically, the one you are suggesting. Will implement this check, so that Popup won't be closed on click inside the popup area